### ISSUE TITLE
Test suite failures in grepmail 5.3034

### ISSUE THREAD

**Issue by GoogleCodeExporter** [coppit/grepmail#2]
```
This is on Fedora Linux development branch but I don't think that's a
factor really.

With Mail::Mbox::MessageParser 1.5000 I get this failure:

$ /usr/bin/make test
PERL_DL_NONLAZY=1 /usr/bin/perl "-MExtUtils::Command::MM" "-e"
"test_harness(0, 'inc', 'blib/lib', 'blib/arch')" t/*.t
t/append_header.t ........... ok
t/auto_search.t ............. ok
t/body.t .................... ok
t/cache.t ................... ok
t/complex_expression.t ...... ok
t/count.t ................... ok
t/date.t .................... ok
t/date_manip.t .............. ok
t/header.t .................. ok
t/help.t .................... ok
t/ignore_signatures.t ....... ok
t/invalid_date.t ............ ok
t/invalid_mailbox.t ......... ok
t/line_number.t ............. ok
t/list.t .................... ok
t/match_compressed.t ........ ok
t/match_headers.t ........... ok
t/match_words.t ............. ok
t/nonexistent_mailbox.t ..... ok
t/not_match_compressed.t .... ok
t/not_match_uncompressed.t .. ok
t/pattern_file.t ............ ok
t/pipe_compressed.t ......... ok
t/pipe_uncompressed.t ....... ok
t/received_date.t ........... ok
#   Failed test 't/temp/recursive_recursive2.stdout compared to
t/results/recursive2'
#   at t/recursive.t line 120.
#     Structures begin differing at:
#          $got->[103] = Does not exist
#     $expected->[103] = 'From dakott@alpha.delta.edu  Thu Jan  1 05:56:53 1998
#     '
# Looks like you failed 1 test of 8.
t/recursive.t ...............
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/8 subtests
t/size.t .................... ok
t/status.t .................. ok
t/unique.t .................. ok
Test Summary Report
-------------------
t/recursive.t             (Wstat: 256 Tests: 8 Failed: 1)
  Failed test:  7
  Non-zero exit status: 1
Files=29, Tests=257, 27 wallclock secs ( 0.26 usr  0.12 sys + 21.18 cusr 
5.62 csys = 27.18 CPU)
Result: FAIL
Failed 1/29 test programs. 1/257 subtests failed.

After updating to Mail::Mbox::MessageParser 1.5002 I get more failures:

 /usr/bin/make test
PERL_DL_NONLAZY=1 /usr/bin/perl "-MExtUtils::Command::MM" "-e"
"test_harness(0, 'inc', 'blib/lib', 'blib/arch')" t/*.t
t/append_header.t ........... ok
t/auto_search.t ............. ok
t/body.t .................... ok
t/cache.t ................... ok
t/complex_expression.t ...... ok
t/count.t ................... ok
t/date.t .................... ok
t/date_manip.t .............. ok
t/header.t .................. ok
t/help.t .................... ok
t/ignore_signatures.t ....... ok
t/invalid_date.t ............ ok
t/invalid_mailbox.t ......... ok
t/line_number.t ............. ok
t/list.t .................... ok
t/match_compressed.t ........ ok
t/match_headers.t ........... ok
t/match_words.t ............. ok
#   Failed test 'Did not encounter an error executing the test when one was
expected.
#
# '
#   at t/nonexistent_mailbox.t line 102.
#   Failed test 'Did not encounter an error executing the test when one was
expected.
#
# '
#   at t/nonexistent_mailbox.t line 102.
# Looks like you planned 8 tests but ran 6.
# Looks like you failed 2 tests of 6 run.
t/nonexistent_mailbox.t .....
Dubious, test returned 2 (wstat 512, 0x200)
Failed 4/8 subtests
t/not_match_compressed.t .... ok
t/not_match_uncompressed.t .. ok
t/pattern_file.t ............ ok
t/pipe_compressed.t ......... ok
t/pipe_uncompressed.t ....... ok
t/received_date.t ........... ok
#   Failed test 't/temp/recursive_recursive2.stdout compared to
t/results/recursive2'
#   at t/recursive.t line 120.
#     Structures begin differing at:
#          $got->[103] = Does not exist
#     $expected->[103] = 'From dakott@alpha.delta.edu  Thu Jan  1 05:56:53 1998
#     '
# Looks like you failed 1 test of 8.
t/recursive.t ...............
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/8 subtests
t/size.t .................... ok
t/status.t .................. ok
t/unique.t .................. ok
Test Summary Report
-------------------
t/nonexistent_mailbox.t   (Wstat: 512 Tests: 6 Failed: 2)
  Failed tests:  1-2
  Non-zero exit status: 2
  Parse errors: Bad plan.  You planned 8 tests but ran 6.
t/recursive.t             (Wstat: 256 Tests: 8 Failed: 1)
  Failed test:  7
  Non-zero exit status: 1
Files=29, Tests=255, 27 wallclock secs ( 0.25 usr  0.12 sys + 21.24 cusr 
5.36 csys = 26.97 CPU)
Result: FAIL
Failed 2/29 test programs. 3/255 subtests failed.

These additional errors are I believe due to the newly-changed behaviour of
Mail::Mbox::MessageParser for empty mailboxes and thus intentional but I
may be wrong about that.

```

Original issue reported on code.google.com by `mcivta.a...@googlemail.com` on 1 Sep 2009 at 2:31

**Comment by pghmcfc** [comment]
I'm still needing to revert the change to t/recursive.t between 5.3033 and 5.3034 to get the grepmail 5.3100 to build with Mail-Mbox-MessageParser 1.5100.

The empty file issue no longer seems to be happening.

**Comment by coppit** [comment]
Paul, I'll send you private email with a new version that you can test, if you want. But I'm pretty sure these issues are addressed now.

I've repro'd the recursive.t failure, improved the test, and verified the fix. It will be in the version after 5.3100.

The nonexistent_mailbox.t bug was fixed in 5.3100, which didn't make it to CPAN due to a distribution mistake on my part.

Closing.
